### PR TITLE
[QNN EP] Add rank-6 support for Gather, Transpose and Reshape

### PR DESCRIPTION
**Pull request by qti-ashwshan**
### Description
- Node group fusion to support Gather->Transpose->Reshape pattern that handles rank 6 tensors
- Add unit test to ensure this pattern runs on the NPU

### Motivation and Context
- Support rank-6-dimension tensors as QNN does not natively support it.

### CODE REVIEW

File: onnxruntime/core/providers/qnn/builder/qnn_node_group/gather_transpose_reshape_fusion.cc [hunk 1]
```diff
@@ -0,0 +1,543 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (qti-yuduo) [inline onnxruntime/core/providers/qnn/builder/qnn_node_group/gather_transpose_reshape_fusion.cc]: Kyle mentioned we should use our copyright for new files.

File: onnxruntime/test/providers/qnn/qnn_node_group/gather_transpose_reshape_fusion_test.cc [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+  // Run Test
+  RunQnnModelTest(BuildGatherTransposeReshapeTestCase(input_def, indices_shape, indices_data, transpose_perm, final_shape),
+                  provider_options,
+                  13,  // opset
+                  ExpectedEPNodeAssignment::All,
+                  1e-2f);  // fp32_abs_err: HTP uses fixed-point arithmetic, allow small numerical differences
+}
+
+// Test Case 3: Invalid Fusion - Transpose perm violates constraints
```
Comment (qti-yuduo) [inline onnxruntime/test/providers/qnn/qnn_node_group/gather_transpose_reshape_fusion_test.cc]: Remove?

File: onnxruntime/core/providers/qnn/builder/qnn_node_group/gather_transpose_reshape_fusion.cc [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+  RETURN_IF_NOT(qnn_model_wrapper.GetOnnxShape(gather_input.shape, x_shape),
+                "GatherTransposeReshapeFusion: failed to get gather input shape");
+  RETURN_IF_NOT(x_shape.size() == kRank5,
+                "GatherTransposeReshapeFusion: gather input must be rank-5");
+
+  const uint32_t d0 = x_shape[0];
+  const uint32_t d1 = x_shape[1];
+  const uint32_t d2 = x_shape[2];
+  const uint32_t d3 = x_shape[3];
+  const uint32_t merged = d0 * d1 * d2;
```
Comment (qti-yuduo) [inline onnxruntime/core/providers/qnn/builder/qnn_node_group/gather_transpose_reshape_fusion.cc:208]: Things are static here, let's add a bound check (< `UINT32_MAX`) to be safe.

File: onnxruntime/core/providers/qnn/builder/qnn_node_group/gather_transpose_reshape_fusion.cc [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+  const uint32_t u_idx0 = static_cast<uint32_t>(idx0);
+  const uint32_t u_idx1 = static_cast<uint32_t>(idx1);
+
+  const uint32_t factor0 = (index_pattern == GatherIndicesPattern::kColMajor) ? u_idx1 : u_idx0;
+  const uint32_t factor1 = (index_pattern == GatherIndicesPattern::kColMajor) ? u_idx0 : u_idx1;
+
+  const std::string base = utils::GetUniqueName(*gather);
+  const std::string reshape1_out_name = base + "_gtr_r1_out";
```
Comment (qti-yuduo) [inline onnxruntime/core/providers/qnn/builder/qnn_node_group/gather_transpose_reshape_fusion.cc]: `utils::GetUniqueName` support suffix as 2nd argument.

**Review by qti-yuduo — COMMENTED** [review]
Let's do few more iteration with AI so that we have:
- Better test coverages on corner cases
- Less verbose comments, naming and coding style should be more self-explanatory

File: onnxruntime/core/providers/qnn/builder/qnn_node_group/gather_transpose_reshape_fusion.cc (hunk 1 above)
Comment (qti-ashwshan) [reply onnxruntime/core/providers/qnn/builder/qnn_node_group/gather_transpose_reshape_fusion.cc]: Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
SPDX-License-Identifier: MIT

Is this the new copyright ?

**Review by qti-yuduo — APPROVED** [review]
